frontend/src/components/pages/volunteer/posting/VolunteerPostingsPage.tsx

```diff
@@ -1,11 +1,240 @@
+      id
+      branch {
+        id
+        name
+      }
+      shifts {
+        id
+        postingId
+        startTime
+        endTime
+      }
+      skills {
+        id
+        name
+      }
+      employees {
+        id
+        branchId
+      }
+      title
+      type
+      status
+      description
+      startDate
+      endDate
+      autoClosingDate
+      numVolunteers
+    }
+  }
+`;
+
+const MS_PER_WEEK = 604800000;
 
 const VolunteerPostingsPage = (): React.ReactElement => {
+  const [postings, setPostings] = useState<PostingResponseDTO[] | null>(null);
+  const [unfilteredPostings, setUnfilteredPostings] = useState<
+    PostingResponseDTO[] | null
+  >(null);
+  const [filter, setFilter] = useState<string>("week");
+
+  useQuery(POSTINGS, {
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data) => {
+      setPostings(data.postings);
+      setUnfilteredPostings(data.postings);
+    },
+  });
+
+  // should posting.startDate type be declared as string instead on frontend?
+  const dateCompareUtil = (start: Date, filterType: string): boolean => {
+    const startDate = new Date(start);
+    return (
+      startDate.getTime() - Date.now() > 0 &&
+      startDate.getTime() - Date.now() <
+        (filterType === "week" ? MS_PER_WEEK : MS_PER_WEEK * 4)
+    );
+  };
+
+  useEffect(() => {
+    let filteredPostings;
+    switch (filter) {
+      case "month":
+        filteredPostings = unfilteredPostings?.filter((posting) =>
+          dateCompareUtil(posting.startDate, "month"),
+        );
+        setPostings(filteredPostings ?? null);
+        break;
+      case "all":
+        setPostings(unfilteredPostings);
+        break;
+      default:
+        filteredPostings = unfilteredPostings?.filter((posting) =>
+          dateCompareUtil(posting.startDate, "week"),
+        );
+        setPostings(filteredPostings ?? null);
+        break;
+    }
+  }, [filter, unfilteredPostings]);
+
+  const changeFilter = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    if (e.target.value) {
+      setFilter(e.target.value);
+    } else {
+      setFilter("week");
+    }
+  };
+
+  // need to refactor this function based on definition of what is an opportunity / event
+  const isVolunteerOpportunity = (start: Date, end: Date): boolean => {
+    const startDate = new Date(start);
+    const endDate = new Date(end);
+
+    return endDate.getTime() - startDate.getTime() > 1000 * 60 * 60 * 24;
+  };
+
+  const volunteerOpportunities = postings?.filter((posting) =>
+    isVolunteerOpportunity(posting.startDate, posting.endDate),
+  );
+  const events = postings?.filter(
+    (posting) => !isVolunteerOpportunity(posting.startDate, posting.endDate),
+  );
+
   return (
-    <div>
-      <Text textStyle="display-large">Volunteer Postings</Text>
-    </div>
+    <Box bg="background.light" pt="48px" minHeight="100vh">
+      <Box maxW="1280px" mx="auto">
+        <HStack justify="space-between" bg="#F4F4F4" pb="36px">
```

Comment:
This should be the same as background.light which is specified for the enclosing Box

```suggestion
        <HStack justify="space-between" pb="26px">
```